meilisearch-http/src/routes/swap_indexes.rs

```diff
@@ -0,0 +1,131 @@
+        if !search_rules.is_index_authorized(&lhs) {
+            unknown_indexes.insert(lhs.clone());
+        }
+        if !search_rules.is_index_authorized(&rhs) {
+            unknown_indexes.insert(rhs.clone());
+        }
+
+        swaps.push((lhs.clone(), rhs.clone()));
+
+        let is_unique_index_lhs = indexes_set.insert(lhs.clone());
+        if !is_unique_index_lhs {
+            duplicate_indexes.insert(lhs);
+        }
+        let is_unique_index_rhs = indexes_set.insert(rhs.clone());
+        if !is_unique_index_rhs {
+            duplicate_indexes.insert(rhs);
+        }
+    }
+    if !duplicate_indexes.is_empty() {
+        return Err(DuplicateSwappedIndexError {
+            indexes: duplicate_indexes.into_iter().collect(),
+        }
+        .into());
+    }
+    if !unknown_indexes.is_empty() {
+        return Err(IndexesNotFoundError { indexes: unknown_indexes.into_iter().collect() }.into());
+    }
+
+    let task = KindWithContent::IndexSwap { swaps };
+
+    let task = index_scheduler.register(task)?;
+    let task_view = TaskView::from_task(&task);
+
+    Ok(HttpResponse::Accepted().json(task_view))
+}
+
+pub mod errors {
+    use std::fmt::Display;
+
+    use meilisearch_types::error::{Code, ErrorCode};
+
+    #[derive(Debug)]
+    pub struct IndexesNotFoundError {
+        pub indexes: Vec<String>,
+    }
+    impl Display for IndexesNotFoundError {
+        fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+            if self.indexes.len() == 1 {
+                write!(f, "Index `{}` not found,", self.indexes[0])?;
+            } else {
+                write!(f, "Indexes `{}`", self.indexes[0])?;
+                for index in self.indexes.iter().skip(1) {
+                    write!(f, ", `{}`", index)?;
+                }
+                write!(f, "not found.")?;
+            }
+            Ok(())
+        }
+    }
+    impl std::error::Error for IndexesNotFoundError {}
+    impl ErrorCode for IndexesNotFoundError {
+        fn error_code(&self) -> Code {
+            Code::IndexNotFound
+        }
+    }
+    #[derive(Debug)]
+    pub struct DuplicateSwappedIndexError {
+        pub indexes: Vec<String>,
+    }
+    impl Display for DuplicateSwappedIndexError {
+        fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+            if self.indexes.len() == 1 {
+                write!(f, "Indexes must be declared only once during a swap. `{}` was specified several times.", self.indexes[0])?;
+            } else {
+                write!(
+                    f,
+                    "Indexes must be declared only once during a swap. `{}`",
+                    self.indexes[0]
+                )?;
+                for index in self.indexes.iter().skip(1) {
+                    write!(f, ", `{}`", index)?;
+                }
+                write!(f, "were specified several times.")?;
+            }
+
+            Ok(())
+        }
+    }
```

Comment:
Same it would be nice if we could use our already defined error type with maybe two more variants to handle both cases?